core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -115,8 +115,8 @@ class SocketServer(val config: KafkaConfig,
   private var stopped = false
 
   // Socket server metrics
-  newGauge(s"${DataPlaneAcceptor.MetricPrefix}NetworkProcessorAvgIdlePercent", () => SocketServer.this.synchronized {
-    val dataPlaneProcessors = dataPlaneAcceptors.asScala.values.flatMap(a => a.processors)
+  newGauge(s"${DataPlaneAcceptor.MetricPrefix}NetworkProcessorAvgIdlePercent", () => {
+    val dataPlaneProcessors = dataPlaneAcceptors.asScala.values.flatMap(a => a.processors).toSeq
```

Comment:
convert it to seq in order to make sure the "size" and "isEmpty" are consistency

Reply:
Maybe this comment would be useful in the code itself?

Reply:
> Maybe this comment would be useful in the code itself?

you are right :)